modules/Core/classes/Avatars/UploadedImageAvatarSource.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+class UploadedImageAvatarSource extends AvatarSourceBase {
+
+    public function __construct(Language $language) {
+        $this->_name = $language->get('admin', 'avatar_source_uploaded_image');
+        $this->_module = 'Core';
+        $this->_settings = ROOT_PATH . '/modules/Core/includes/admin_avatar_settings/uploaded_image.php';
+    }
+
+    public function get(User $user): ?string {
+        $base_url = ($this->_full_url ? rtrim(URL::getSelfURL(), '/') : '') . ((defined('CONFIG_PATH')) ? CONFIG_PATH . '/' : '/') . 'uploads/avatars';
+
+        if (Settings::get('custom_user_avatars') && $user->data()->has_avatar) {
```

Comment:
Allow custom user avatars? setting on custom avatar source does not save on enabled state